aws/resource_aws_pinpoint_app.go

```diff
@@ -146,9 +154,14 @@ func resourceAwsPinpointAppCreate(d *schema.ResourceData, meta interface{}) erro
 
 	log.Printf("[DEBUG] Pinpoint create app: %s", name)
 
+	if v, ok := d.GetOk("tags"); ok {
+		tags = tagsFromMapPinPointApp(v.(map[string]interface{}))
```

Comment:
Instead of creating a new variable to store the tags, this logic can be moved below the `CreateAppInput` declaration and call `req.CreateApplicationRequest.Tags = ` directly.

```go
	req := &pinpoint.CreateAppInput{
		CreateApplicationRequest: &pinpoint.CreateApplicationRequest{
			Name: aws.String(name),
		},
	}

	if v, ok := d.GetOk("tags"); ok {
		req.CreateApplicationRequest.Tags = tagsFromMapPinPointApp(v.(map[string]interface{}))
	}
```

Reply:
Make sense, Implemented 👍